be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/DbActor.scala

```diff
@@ -137,157 +180,269 @@ object DbActor extends AskPatternConstants {
-        )
-      }
-
+      //FIXME: check if this should be kept
       super.preStart()
     }
 
-    private def createDatabase(channel: Channel): DB = {
-      val options: Options = new Options()
-      options.createIfMissing(true)
-
-      val channelName: String = s"$DATABASE_FOLDER$channel"
-      val channelDb = factory.open(new File(channelName), options)
-
-      channelNamesDb.put(channelName.getBytes, "".getBytes)
-      channelsMap += (channel -> channelDb)
-
-      channelDb
-    }
-
+    // only for the messages
     private def write(channel: Channel, message: Message): DbActorMessage = {
-      val channelDb: DB = channelsMap.get(channel) match {
-        case Some(channelDb) => channelDb
-        case _ => createDatabase(channel)
+      
+      //encapsulate this inside function
+      Try(db.get(channel.toString.getBytes)) match {
+        case Success(bytes) if bytes != null =>
+          log.info(s"Channel '$channel' already exists.")
+        case _ =>
+          //FIXME: handle type of message (is it actually a problem or is the MessageExample coded in a weird way?)
+          //val objectType = message.decodedData.getOrElse(null)._object //or some other way, then match using match case and establish channel type
+          val objectType = ObjectType.LAO
+          val actualObjectType = if (objectType == ObjectType.ELECTION || objectType == ObjectType.CHIRP){
+            objectType
+          } else {
+            ObjectType.LAO
+          }
+          createChannel(channel, actualObjectType) match {
+            case DbActorAck() =>
+            case _ =>
+              log.info(s"Actor $self (db) encountered a problem while creating channel '$channel'")
+              DbActorNAck(ErrorCodes.SERVER_ERROR.id, s"Error when creating channel '$channel'")
+          }
       }
-
+      
       val messageId: Hash = message.message_id
-      Try(channelDb.put(messageId.getBytes, message.toJsonString.getBytes)) match {
-        case Success(_) =>
-          log.info(s"Actor $self (db) wrote message_id '$messageId' on channel '$channel'")
-          DbActorWriteAck
+      Try(db.get(channel.toString.getBytes)) match {
+        case Success(bytes) if bytes != null => {
+          val key: String = channel + "/" + messageId.toString
+          Try(db.createWriteBatch()) match {
+            case Success(batch) => {
+              val json = new String(bytes, StandardCharsets.UTF_8)
+              batch.put(channel.toString.getBytes, ChannelData.buildFromJson(json).addMessage(messageId).toJsonString.getBytes)
+              batch.put(key.getBytes, message.toJsonString.getBytes)
+              //allows writing all data atomically
+              Try(db.write(batch)) match {
+                case Success(_) =>
+                  log.info(s"Actor $self (db) wrote object 'objectId' and message_id '$messageId' on channel '$channel'") //change with object and objectId/name/smth like that
```

Comment:
This means the new implemented Object types should have an ID. 
`LaoData `and `ChannelData `don't for now !

Reply:
I was just thinking about this, I believe the best way would be to simply add some kind of "getObjectString()" method in the supertype of LaoData, ElectionData, etc. once I change this method and implement these types so that it can also return ElectionData (for example) and other data so the log sees which type of data was being accessed